core/Vector.carp

```diff
@@ -253,37 +273,35 @@
   (doc normalize "Normalize a vector.")
   (defn normalize [o]
     (let [m (mag o)]
-      (if (= m 0.0)
+      (if (zero? m)
         @o
         (div o m))))
 
-  (doc dot "Get the dot product of the two vectors x and y.")
-  (defn dot [x y]
-    (Maybe.apply (zip * x y)
-                 (fn [x] (Array.reduce &(fn [x y] (+ x @y)) 0.0 (v &x)))))
-
-  (doc angle-between "Get the angle between to vectors a and b.")
+  (doc angle-between "Get the angle between two vectors a and b.")
   (defn angle-between [a b]
-    (Maybe.apply (dot a b)
+    (Maybe.apply (VectorN.dot a b)
                  (fn [x]
-                    (let [dmm (/ x (* (mag a) (mag b)))]
-                      (Double.acos (Double.clamp -1.0 1.0 dmm))))))
+                    (let [dmm (/ x (* (VectorN.mag a) (VectorN.mag b)))]
+                      (acos (clamp--1-1 dmm))))))
 
   (doc anti-parallel? "Check whether the two vectors a and b are anti-parallel.")
   (defn anti-parallel? [a b]
-    (Maybe.apply (angle-between a b) (fn [x] (= x Double.pi))))
+    (Maybe.apply (angle-between a b) (fn [x] (= x pi))))
 
   (doc parallel? "Check whether the two vectors a and b are parallel.")
   (defn parallel? [a b]
-    (Maybe.apply (angle-between a b) (fn [x] (= x 0.0))))
+    (Maybe.apply (angle-between a b) (fn [x] (zero? x))))
 
   (doc perpendicular? "Check whether the two vectors a and b are perpendicular.")
   (defn perpendicular? [a b]
-    (Maybe.apply (angle-between a b) (fn [x] (= x (/ Double.pi 2.0)))))
+    (Maybe.apply (angle-between a b) (fn [x] (= x (half-pi)))))
+
+  (doc vlerp "Linearly interpolate between the two vectors a and b by amnt (between 0 and 1).")
+  (defn vlerp [a b amnt]
+    (zip (fn [a b] (lerp a b amnt)) a b))
 
-  (doc lerp "Linearly interpolate between the two vectors a and b by amnt (between 0 and 1).")
-  (defn lerp [a b amnt]
-    (Maybe.apply (zip - b a)
-      (fn [x] (init @(n a) @(v &(zip- * &(Array.replicate @(n a) &amnt)
-                           (v &x)))))))
+  (defn workaround-516 []
```

Comment:
Note this horrible workaround for #516, this should be removed when that one gets fixed.

Reply:
It doesn’t seem like this code is used anywhere. Am I reading this right as just being there to appease the compiler?

If so, this is extremely weird.

Reply:
Just remove that and try to run `test/vector3.carp`, you'll hit #516. I tried to fix it but my limited knowledge of the compiler internals didn't help.